apps/processing/src/services/sharedDependencies.service.ts

```diff
@@ -68,8 +74,16 @@ export class SharedDependenciesService {
         const eventsRegistry = new InMemoryEventsRegistry(
             new Logger({ className: "InMemoryEventsRegistry" }),
         );
-        const strategyRegistry = new InMemoryStrategyRegistry(
-            new Logger({ className: "InMemoryStrategyRegistry" }),
+        const strategyRepository = new KyselyStrategyRepository(
+            kyselyDatabase,
+            env.DATABASE_SCHEMA,
+        );
+        const strategyRegistry = await InMemoryCachedStrategyRegistry.initialize(
+            new Logger({ className: "InMemoryCachedStrategyRegistry" }),
+            new DatabaseStrategyRegistry(
+                new Logger({ className: "DatabaseStrategyRegistry" }),
```

Comment:
I am a little concerned about using multiple logger instances. I think it adds tons of overhead, think about a logger that sends logs to a server or prints all the logs into a file, for each instance you will have a connection (1st case) or a dedicated file (2nd case) correct me if iam wrong pls.

i would love to hear @0xyaco 's opinion about having multiple logger instances across the application

Reply:
mmm i see what you're pointing, we can better discuss it on the Error handling and Logging milestone but a solution could be a Context object on each call so we know which file or chain emitted a log?